docs/content/guides/2.alokai-essentials/1.alokai-next-js/9.udl.md

```diff
@@ -0,0 +1,508 @@
+
+```ts
+require('dotenv').config();
+import {
+  Config,
+  Context,
+  createUnifiedExtension,
+  methods,
+  normalizers
+} from "@vsf-enterprise/unified-api-sapcc";
+import { ApiClientExtension } from "@vue-storefront/middleware";
+
+const apiMethods = methods<typeof normalizers>();
+export const unifiedApiExtension = createUnifiedExtension<Context, Config>()({
+  normalizers,
+  apiMethods,
+  config: {
+    transformImageUrl: (url: string) => {
+      return new URL(url, process.env.SAPCC_BASE_URL).toString();
+    },
+    defaultCurrency: "USD",
+  }
+});
+
+export const integrations = {
+  commerce: {
+    location: '@vsf-enterprise/sapcc-api/server',
+    configuration: {
+      OAuth: {
+        uri: process.env.SAPCC_OAUTH_URI,
+        clientId: process.env.SAPCC_OAUTH_CLIENT_ID,
+        clientSecret: process.env.SAPCC_OAUTH_CLIENT_SECRET,
+        tokenEndpoint: process.env.SAPCC_OAUTH_TOKEN_ENDPOINT,
+        tokenRevokeEndpoint: process.env.SAPCC_OAUTH_TOKEN_REVOKE_ENDPOINT,
+        cookieOptions: {
+          'vsf-sap-token': { secure: process.env.NODE_ENV !== 'development' }
+        }
+      },
+      api: {
+        uri: process.env.SAPCC_API_URI,
+        baseSiteId: process.env.DEFAULT_BASE_SITE_ID,
+        catalogId: process.env.DEFAULT_CATALOG_ID,
+        catalogVersion: process.env.DEFAULT_CATALOG_VERSION,
+        defaultLanguage: process.env.DEFAULT_LANGUAGE,
+        defaultCurrency: process.env.DEFAULT_CURRENCY
+      }
+    },
+    extensions: (extensions: ApiClientExtension[]) => [...extensions, unifiedApiExtension]
+  }
+};
+```
+
+Next, let's prepare Unified API Extension for the Alokai SDK. Create a new file inside `middleware` directory called `types.ts` and add the following code:
+
+```ts
+import { WithoutContext } from "@vue-storefront/middleware";
+import { unifiedApiExtension } from "./middleware.config";
+
+export type UnifiedApiExtension = typeof unifiedApiExtension;
+export type UnifiedEndpoints = WithoutContext<UnifiedApiExtension["extendApiMethods"]>;
+```
+
+This file exports two types: `UnifiedApiExtension` and `UnifiedEndpoints`. This is needed for SDK to automatically generate type safe methods based on Unified `apiMethods` object.
+
+Great! Now we have successfully installed and configured the Unified Data Layer in our Alokai Middleware. Next, we need to configure Alokai SDK to use the Unified Data Layer.
+
+### Configuring Alokai SDK
+
+In order to configure Alokai SDK with Unified Data Layer, we need to change the SDK configuration. Open `storefront/sdk/sdk.config.ts` file and add the following code:
+
+```diff
+- import { sapccModule } from '@vsf-enterprise/sapcc-sdk';
+import { CreateSdkOptions, createSdk } from "@vue-storefront/next";
++import type { UnifiedEndpoints } from 'middleware/types';
+
+export const { getSdk } = createSdk(
+  options,
+  ({ buildModule, middlewareModule, middlewareUrl, getRequestHeaders }) => ({
+-    sapcc: buildModule(sapccModule, {
```

Comment:
It's not relevant but this diff is outdated - in 4.install-sdk.md `middlewareModule` was already used